prac_09/taxi_simulator.py

```diff
@@ -0,0 +1,82 @@
+from prac_09.taxi import Taxi
+from prac_09.silver_service_taxi import SilverServiceTaxi
+
+
+def display_taxis(taxis):
+    """Display all available taxis with their details."""
+    print("Taxis available:")
+    for i, taxi in enumerate(taxis):
+        print(f"{i} - {taxi}")
+
+
+def choose_taxi(taxis):
+    """Prompt the user to choose a taxi and return the selected taxi, or None if invalid."""
+    display_taxis(taxis)
+    try:
+        taxi_choice = int(input("Choose taxi: "))
+        if 0 <= taxi_choice < len(taxis):
+            return taxis[taxi_choice]
+        else:
+            print("Invalid taxi choice")
+    except ValueError:
+        print("Invalid input; please enter a number.")
+    return None
+
+
+def drive_taxi(current_taxi, total_cost):
+    """Drive the selected taxi, calculate the fare, and update the total cost."""
+    try:
+        distance = int(input("Drive how far? "))
+        actual_distance = current_taxi.drive(distance)
+        trip_cost = current_taxi.get_fare()
+        total_cost += trip_cost
+        print(f"Your {current_taxi.name} trip cost you ${trip_cost:.2f}")
+        current_taxi.start_fare()  # Reset fare distance for the next trip
+    except ValueError:
+        print("Invalid input; please enter a number.")
+    return total_cost
+
+
+def main():

```

Comment:
Is main() meant to be this far down? Perhaps it would be clearer if main was at the top? So, at first glance, other programmers know what the program is?